pkg/spiceerrors/common.go

```diff
@@ -51,3 +54,35 @@ func AsErrorWithSource(err error) (*ErrorWithSource, bool) {
 	}
 	return nil, false
 }
+
+// ErrorWithAdditionalDetails is an error that includes additional details.
+type ErrorWithAdditionalDetails struct {
+	error
+
+	// AdditionalDetails is a map of additional details for the error.
+	AdditionalDetails map[string]string
+}
+
+func NewErrorWithAdditionalDetails(err error) *ErrorWithAdditionalDetails {
+	return &ErrorWithAdditionalDetails{err, nil}
+}
+
+// Unwrap returns the inner, wrapped error.
+func (err *ErrorWithAdditionalDetails) Unwrap() error {
+	return err.error
+}
+
+func (err *ErrorWithAdditionalDetails) WithAdditionalDetails(key string, value string) {
+	if err.AdditionalDetails == nil {
+		err.AdditionalDetails = make(map[string]string)
+	}
+	err.AdditionalDetails[key] = value
+}
+
+func (err *ErrorWithAdditionalDetails) Details(details map[string]string) map[string]string {
```

Comment:
I don't think the name of the function clarifies what it does. Maybe `Merge`?

Reply:
Details is what it is called on the underlying gRPC error. I could make it `MergeDetails` or `AddToDetails`?

Reply:
either works for me

Reply:
Updated